### PR TITLE
Use generic types in HDF5 type guards

### PR DESCRIPTION
**Pull request by axelboc**
Follow up to #326

### CODE REVIEW

File: src/h5web/metadata-viewer/EntityInfo.tsx [hunk 1]
```diff
@@ -24,17 +24,21 @@ function EntityInfo(props: Props): ReactElement {
...
-        {'type' in entity && (
+        {(isDataset(entity) || isDatatype(entity)) && (
           <tr>
             <th scope="row">Type</th>
-            <td>{isBaseType(entity.type) ? entity.type.class : 'Complex'}</td>
+            <td>
+              {typeof entity.type === 'string'
+                ? entity.type
+                : entity.type.class}
+            </td>
```
Comment (axelboc) [inline src/h5web/metadata-viewer/EntityInfo.tsx:34]: The term `Complex` was confusing and the condition was hiding useful information.

File: src/h5web/metadata-viewer/EntityInfo.tsx [hunk 2]
```diff
@@ -24,17 +24,21 @@ function EntityInfo(props: Props): ReactElement {
         </tr>
       </thead>
       <tbody>
-        {'type' in entity && (
+        {(isDataset(entity) || isDatatype(entity)) && (
           <tr>
             <th scope="row">Type</th>
-            <td>{isBaseType(entity.type) ? entity.type.class : 'Complex'}</td>
+            <td>
+              {typeof entity.type === 'string'
```
Comment (loichuder) [inline src/h5web/metadata-viewer/EntityInfo.tsx:31]: This condition puzzles me.

`entity.type` can be `string` or something that has a `class` ?

**Review by loichuder — APPROVED** [review]
Looks nicer !

File: src/h5web/metadata-viewer/EntityInfo.tsx (hunk 2 above)
Comment (axelboc) [reply src/h5web/metadata-viewer/EntityInfo.tsx:31]: Yeah, it can be the ID of a datatype entity.

**Pull request merged**